pandas/core/arrays/string_.py

```diff
@@ -198,16 +198,19 @@ def __from_arrow__(
                 # pyarrow.ChunkedArray
                 chunks = array.chunks
 
-            results = []
-            for arr in chunks:
-                # using _from_sequence to ensure None is converted to NA
-                str_arr = StringArray._from_sequence(np.array(arr))
-                results.append(str_arr)
-
-        if results:
-            return StringArray._concat_same_type(results)
+        if len(chunks) == 0:
+            arr = np.array([], dtype=object)
         else:
-            return StringArray(np.array([], dtype="object"))
+            arr = pyarrow.concat_arrays(chunks).to_numpy(zero_copy_only=False)
+            arr = lib.convert_nans_to_NA(arr)
+        # Bypass validation inside StringArray constructor
```

Comment:
Is the validation really that expensive? IMO I think going through the constructor normally is a lot clearer than calling `__new__` and `__init__` manually

Reply:
Yes, it is - 
```python
n = 10000000
df = pd.DataFrame(
    {
        "a": np.random.choice(["a", "b", None], n),
        "b": np.random.choice(["b", "c", None], n),
        "c": np.random.choice(["c", "d", None], n),
        "d": np.random.choice(["d", "e", None], n),
    },
    dtype=object,
)
df = df.convert_dtypes()
table = pa.Table.from_pandas(df)
# Remove metadata to prevent automatic use of nullable types
table = table.replace_schema_metadata()
pq.write_table(table, "test.parquet")

%time df = pd.read_parquet("test.parquet", use_nullable_dtypes=True)
```

On 1.5.1:
```
: CPU times: user 2.8 s, sys: 264 ms, total: 3.07 s
: Wall time: 2.46 s
```
On this branch:
```
: CPU times: user 1.37 s, sys: 113 ms, total: 1.49 s
: Wall time: 923 ms
```

When using `use_nullable_types=False`, we get:
```
: CPU times: user 1.22 s, sys: 133 ms, total: 1.35 s
: Wall time: 744 ms
```